### PR TITLE
End point security EPG selector resources

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

File: aci/resource_aci_fvepgselector.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"relation_fv_rs_match_e_pg": {
```
Comment (lhercot) [inline aci/resource_aci_fvepgselector.go:43]: relation_fv_rs_match_e_pg -> relation_fv_rs_match_epg

Comment (lhercot) [reply aci/resource_aci_fvepgselector.go:43]: What is the difference between relation_fv_rs_match_epg and match_epg_dn?

File: aci/resource_aci_fvepgselector.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+	if relationTofvRsMatchEPg, ok := d.GetOk("relation_fv_rs_match_e_pg"); ok {
+		relationParam := relationTofvRsMatchEPg.(string)
+		err = aciClient.CreateRelationfvRsMatchEPg(fvEPgSelector.DistinguishedName, fvEPgSelectorAttr.Annotation, relationParam)
+
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		d.Partial(true)
+		//d.SetPartial("relation_fv_rs_match_e_pg")
+		d.Partial(false)
```
Comment (lhercot) [inline aci/resource_aci_fvepgselector.go:152]: Do we still need those Partial if we don't do anything in between?

File: aci/resource_aci_fvepgselector_test.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccAciEndpointSecurityGroupEPgSelector_Basic(t *testing.T) {
+	var endpoint_security_group_e_pg_selector models.EndpointSecurityGroupEPgSelector
+	fv_tenant_name := acctest.RandString(5)
+	fv_ap_name := acctest.RandString(5)
+	fv_e_sg_name := acctest.RandString(5)
```
Comment (lhercot) [inline aci/resource_aci_fvepgselector_test.go:18]: fv_esg_name and fv_epg_selector_name

File: aci/resource_aci_fvepgselector_test.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	fv_e_pg_selector_name := acctest.RandString(5)
+	description := "endpoint_security_group_e_pg_selector created while acceptance testing"
```
Comment (lhercot) [inline aci/resource_aci_fvepgselector_test.go:20]: _e_pg_ -> _epg_

Comment (lhercot) [reply aci/resource_aci_fvepgselector_test.go:20]: Fix all the other occurence of this and of _e_sg_ -> _esg_

File: examples/aci_esg_epg_selector/aci_esg_epg_selector.tf [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  ip_learning                    = "yes"
+  limit_ip_learn_to_subnets      = "yes"
+  mcast_allow                    = "yes"
+  multi_dst_pkt_act              = "bd-flood"
+  bridge_domain_type             = "regular"
+  unicast_route                  = "yes"
+  unk_mac_ucast_act              = "flood"
+  unk_mcast_act                  = "flood"
+  vmac                           = "not-applicable"
+
```
Comment (lhercot) [inline examples/aci_esg_epg_selector/aci_esg_epg_selector.tf:35]: Remove empty line and run terraform fmt on the file.

File: go.mod [hunk 6]
```diff
@@ -5,5 +5,6 @@ go 1.12
 require (
 	github.com/ciscoecosystem/aci-go-client v1.14.0
 	github.com/ghodss/yaml v1.0.0
+	github.com/hashicorp/terraform-plugin-sdk v1.17.2 // indirect
```
Comment (lhercot) [inline go.mod:8]: Why do we need this? Probably a wrong import somewhere. We should not need this old plugin. Everything should be v2.

File: website/docs/d/endpoint_security_group_epg_selector.html.markdown [hunk 7]
```diff
@@ -0,0 +1,41 @@
+---
+layout: "aci"
+page_title: "ACI: aci_endpoint_security_group_e_pg_selector"
+sidebar_current: "docs-aci-data-source-endpoint_security_group_e_pg_selector"
+description: |-
+  Data source for ACI Endpoint Security Group EPg Selector
+---
+
+# aci_endpoint_security_group_e_pg_selector #
```
Comment (lhercot) [inline website/docs/d/endpoint_security_group_epg_selector.html.markdown:9]: e_pg -> epg

File: website/docs/d/endpoint_security_group_epg_selector.html.markdown [hunk 8]
```diff
@@ -0,0 +1,41 @@
+---
+layout: "aci"
+page_title: "ACI: aci_endpoint_security_group_e_pg_selector"
+sidebar_current: "docs-aci-data-source-endpoint_security_group_e_pg_selector"
+description: |-
+  Data source for ACI Endpoint Security Group EPg Selector
```
Comment (lhercot) [inline website/docs/d/endpoint_security_group_epg_selector.html.markdown:6]: EPg -> EPG

File: website/docs/d/endpoint_security_group_epg_selector.html.markdown [hunk 9]
```diff
@@ -0,0 +1,41 @@
+---
+layout: "aci"
+page_title: "ACI: aci_endpoint_security_group_e_pg_selector"
+sidebar_current: "docs-aci-data-source-endpoint_security_group_e_pg_selector"
```
Comment (lhercot) [inline website/docs/d/endpoint_security_group_epg_selector.html.markdown:4]: e_pg -> epg

File: website/docs/d/endpoint_security_group_epg_selector.html.markdown [hunk 10]
```diff
@@ -0,0 +1,41 @@
+---
+layout: "aci"
+page_title: "ACI: aci_endpoint_security_group_e_pg_selector"
```
Comment (lhercot) [inline website/docs/d/endpoint_security_group_epg_selector.html.markdown:3]: e_pg -> epg

File: website/docs/d/endpoint_security_group_epg_selector.html.markdown [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+sidebar_current: "docs-aci-data-source-endpoint_security_group_e_pg_selector"
+description: |-
+  Data source for ACI Endpoint Security Group EPg Selector
+---
+
+# aci_endpoint_security_group_e_pg_selector #
+
+Data source for ACI Endpoint Security Group EPg Selector
```
Comment (lhercot) [inline website/docs/d/endpoint_security_group_epg_selector.html.markdown:11]: EPG

File: website/docs/d/endpoint_security_group_epg_selector.html.markdown [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+## GUI Information ##
+
+* `Location` - Tenants > {tenant_name} > Application Profiles > Endpoint Security Groups > Selectors > EPG Selectors
+
+
+## Example Usage ##
+
+```hcl
+data "aci_endpoint_security_group_e_pg_selector" "example" {
```
Comment (lhercot) [inline website/docs/d/endpoint_security_group_epg_selector.html.markdown:27]: e_pg -> epg

File: website/docs/r/endpoint_security_group_epg_selector.html.markdown [hunk 13]
```diff
@@ -0,0 +1,47 @@
+---
+layout: "aci"
+page_title: "ACI: aci_endpoint_security_group_e_pg_selector"
+sidebar_current: "docs-aci-resource-endpoint_security_group_e_pg_selector"
+description: |-
+  Manages ACI Endpoint Security Group EPg Selector
```
Comment (lhercot) [inline website/docs/r/endpoint_security_group_epg_selector.html.markdown:6]: EPG

File: website/docs/d/endpoint_security_group_epg_selector.html.markdown [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+  endpoint_security_group_dn  = aci_endpoint_security_group.example.id
+  match_epg_dn  = aci_application_epg.example.id 
+}
+```
+
+## Argument Reference ##
+
+* `endpoint_security_group_dn` - (Required) Distinguished name of parent EndpointSecurityGroup object.
```
Comment (lhercot) [inline website/docs/d/endpoint_security_group_epg_selector.html.markdown:35]: Endpoint Security Group

File: website/docs/r/endpoint_security_group_epg_selector.html.markdown [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+
+## GUI Information ##
+
+* `Location` - Tenants > {tenant_name} > Application Profiles > Endpoint Security Groups > Selectors > EPG Selectors
+
+
+## Example Usage ##
+
+```hcl
+resource "aci_endpoint_security_group_e_pg_selector" "example" {
```
Comment (lhercot) [inline website/docs/r/endpoint_security_group_epg_selector.html.markdown:26]: e_pg -> epg

File: website/docs/r/endpoint_security_group_epg_selector.html.markdown [hunk 16]
```diff
@@ -0,0 +1,47 @@
+---
+layout: "aci"
+page_title: "ACI: aci_endpoint_security_group_e_pg_selector"
+sidebar_current: "docs-aci-resource-endpoint_security_group_e_pg_selector"
+description: |-
+  Manages ACI Endpoint Security Group EPg Selector
+---
+
+# aci_endpoint_security_group_e_pg_selector #
```
Comment (lhercot) [inline website/docs/r/endpoint_security_group_epg_selector.html.markdown:9]: e_pg -> epg

File: website/docs/r/endpoint_security_group_epg_selector.html.markdown [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+
+Manages ACI Endpoint Security Group EPg Selector
```
Comment (lhercot) [inline website/docs/r/endpoint_security_group_epg_selector.html.markdown:11]: EPG

File: website/docs/r/endpoint_security_group_epg_selector.html.markdown [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+resource "aci_endpoint_security_group_e_pg_selector" "example" {
+  endpoint_security_group_dn  = aci_endpoint_security_group.example.id
+  match_epg_dn  = aci_application_epg.example.id 
+  annotation = "orchestrator:terraform"
+}
+```
+
+## Argument Reference ##
+
+* `endpoint_security_group_dn` - (Required) Distinguished name of parent EndpointSecurityGroup object.
```
Comment (lhercot) [inline website/docs/r/endpoint_security_group_epg_selector.html.markdown:35]: Endpoint Security Group

File: website/docs/r/endpoint_security_group_epg_selector.html.markdown [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+* `match_epg_dn` - (Required) EPG Dn to be associated.
+* `annotation` - (Optional) Annotation of object Endpoint Security Group EPg Selector.
```
Comment (lhercot) [inline website/docs/r/endpoint_security_group_epg_selector.html.markdown:37]: EPG

File: website/docs/d/endpoint_security_group_epg_selector.html.markdown [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+* `match_epg_dn` - (Required) EPG Dn to be associated. 
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the Endpoint Security Group EPg Selector.
+* `annotation` - (Optional) Annotation of object Endpoint Security Group EPg Selector.
```
Comment (lhercot) [inline website/docs/d/endpoint_security_group_epg_selector.html.markdown:40]: EPG

File: website/docs/d/endpoint_security_group_epg_selector.html.markdown [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+}
+```
+
+## Argument Reference ##
+
+* `endpoint_security_group_dn` - (Required) Distinguished name of parent EndpointSecurityGroup object.
+* `match_epg_dn` - (Required) EPG Dn to be associated. 
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the Endpoint Security Group EPg Selector.
```
Comment (lhercot) [inline website/docs/d/endpoint_security_group_epg_selector.html.markdown:39]: EPG

File: website/docs/d/endpoint_security_group_epg_selector.html.markdown [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+* `annotation` - (Optional) Annotation of object Endpoint Security Group EPg Selector.
+* `name_alias` - (Optional) Name Alias of object Endpoint Security Group EPg Selector.
```
Comment (lhercot) [inline website/docs/d/endpoint_security_group_epg_selector.html.markdown:41]: EPG

**Review by lhercot — CHANGES_REQUESTED** [review]
Update the aci-go-client to v1.15.0 to include the needed class.

Did you also run successfully the new tests?

File: aci/resource_aci_fvepgselector.go (hunk 2 above)
Comment (shrsr) [reply aci/resource_aci_fvepgselector.go:152]: No. I'll remove it

File: aci/resource_aci_fvepgselector_test.go (hunk 4 above)
Comment (shrsr) [reply aci/resource_aci_fvepgselector_test.go:20]: Done

File: aci/resource_aci_fvepgselector_test.go (hunk 3 above)
Comment (shrsr) [reply aci/resource_aci_fvepgselector_test.go:18]: done

File: website/docs/r/endpoint_security_group_epg_selector.html.markdown [hunk 23]
```diff
@@ -0,0 +1,47 @@
+---
+layout: "aci"
+page_title: "ACI: aci_endpoint_security_group_e_pg_selector"
+sidebar_current: "docs-aci-resource-endpoint_security_group_e_pg_selector"
```
Comment (lhercot) [inline website/docs/r/endpoint_security_group_epg_selector.html.markdown:4]: e_pg ->epg

File: website/docs/r/endpoint_security_group_epg_selector.html.markdown [hunk 24]
```diff
@@ -0,0 +1,47 @@
+---
+layout: "aci"
+page_title: "ACI: aci_endpoint_security_group_e_pg_selector"
```
Comment (lhercot) [inline website/docs/r/endpoint_security_group_epg_selector.html.markdown:3]: e_pg ->epg

**Review by lhercot — CHANGES_REQUESTED** [review]
I had to unresolve some of the previous comments as they are not fixed. Please verify them and fix them or commit the changes if you missed them in your commit.

Also found 2 new ones.

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**